File: 820-short-encoding-of-words/820-short-encoding-of-words.py
class Solution:
    def minimumLengthEncoding(self, words: List[str]) -> int:
        st = set(words)
        
        for word in words:
            for i in range(1, len(word)):
                st.discard(word[i:])
                
        ans = 0
        for i in st:
            ans += len(i)+1
        return ans
        

File: 820-short-encoding-of-words/820-short-encoding-of-words.cpp
class Solution {
public:
    int minimumLengthEncoding(vector<string>& words) {
        if(words.size() == 1) return words[0].size()+1;
        
        unordered_set<string> set(words.begin(), words.end());
        
        for(auto word: set){
            for(int ind=1; ind<word.size(); ind++){
                string s = word.substr(ind);
                if(set.find(s) != set.end()) set.erase(s);
            }
        }
        int n = 0;
        for(auto item: set) n+= item.size();
        
        return n+set.size();
    }
};